libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,657 @@
+        } else {
+            this.items = this.dsItems;
+        }
+        this.stateChanges.next(this.items);
+        this.id = `fdp-list-${nextListId++}`;
+        // using selection Model for multiselect
+        if (this.selectionMode === 'multi') {
+            this.multiSelect = true;
+            this.ariaMultiselectable = true;
+        } else { this.multiSelect = false; }
+        this.selectionModel = new SelectionModel<BaseListItem>(
+            this.multiSelect,
+            this.selectedItems
+        );
+
+        this.selectionModel.changed.subscribe(e => {
+            this.selectedItems = this.selectionModel.selected;
+            const event = new SelectionChangeEvent();
+            event.selectedItems = this.selectedItems;
+            this.stateChanges.next(event);
+            this.selectedItemChange.emit(event);
+
+        });
+
+
+    }
+
+    /** @hidden */
+    /**Keyboard manager on list items, set values when passed via array */
+    ngAfterViewInit(): void {
+
+        this.keyManager = new FocusKeyManager<BaseListItem>(this.ListItems).withWrap();
+        this.ListItems.forEach((item) => {
+            item.navigated = this.navigated;
+            item.navigationIndicator = this.navigationIndicator;
+            item.contentDensity = this.contentDensity;
+            item.selectionMode = this.selectionMode;
+            item.listType = this.listType;
+            item.hasByLine = this.hasByLine;
+            item.noSeperator = this.noSeperator;
+            this.stateChanges.next(item);
+        });
+
+    }
+
+
+    /** @hidden */
+    /**Setting values from list to list items
+     * example:
+     * Does list item has navigation,
+     * should show arrows,
+     * will it be compact mode,
+     * should be in which selection mode
+     * set values when passed via datasource
+     */
+    ngAfterContentInit(): void {
+        this._itemsSubscription = this.ListItems.changes.subscribe((items) => {
+            items.forEach((item) => {
+                item.navigated = this.navigated;
+                item.navigationIndicator = this.navigationIndicator;
+                item.contentDensity = this.contentDensity;
+                item.selectionMode = this.selectionMode;
+                item.listType = this.listType;
+                item.hasByLine = this.hasByLine;
+                item.noSeperator = this.noSeperator;
+                this.stateChanges.next(item);
+            });
+
+        });
+    }
+
+    ngOnDestroy(): void {
+        super.ngOnDestroy();
+
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+        }
+        if (this._dsSubscription) {
+            this._dsSubscription.unsubscribe();
+        }
+        if (this.selectionModel) {
+            this.selectionModel.changed.unsubscribe();
+        }
+        if (this._itemsSubscription) {
+            this._itemsSubscription.unsubscribe();
+        }
+    }
+
+    @Input()
+    get value(): any {
```

Comment:
setters and getters should not be at the bottom

Reply:
In previous reviews also, this concern was raised to align members in order. So after referring angular coding guidelines. I was following the below sequence.

Member Sequence:

@input
Public used in a template
Private
@ViewChild
@ContentChild
@output
Getter and setter
Public functions
Public functions hidden minded with template
Lifecycle
Private functions

For file import:
angular
external
internal

Please do let me know if this is not correct,  I will update accordingly.